### ISSUE TITLE
Support for conversion to `torch.Tensor`

### ISSUE THREAD

**Issue by samvanstroud** [omni-us/jsonargparse#196]
## 🐛 Bug report

I am encountering a problem using the package with Lightning.

When trying to specify an `init_args` which expects a `torch.Tensor` (for example when specifying the weights of a loss function), an error occurs because torch module expects a tensor, not a list.

It should be possible to [register](https://jsonargparse.readthedocs.io/en/stable/index.html#registering-types) pytorch tensors using

```python
def serializer(x):
    return x.tolist()

def deserializer(x):
    return torch.tensor(x)

register_type(torch.Tensor, serializer, deserializer)
```

However, doing so results in an error:

```bash
main: error: Parser key "model": Problem with given class_path "MyModel":
  - Configuration check failed :: Parser key "my_model": Value "Namespace(class_path='MySubModel', init_args=Namespace(..., loss=Namespace(class_path='torch.nn.CrossEntropyLoss', init_args=Namespace(weight=tensor([  7.6200, 104.9100,   1.5600,  20.6500,  14.0500,  14.0600,   1.0000,  36.0700]), size_average=None, ignore_index=-100, reduce=None, reduction='mean', label_smoothing=0.0))))" does not validate against any of the types in typing.Optional[MySubModel]:
    - Boolean value of Tensor with more than one value is ambiguous
...
```

Since a list is specified in the yaml and the namespace shows a `tensor([...])`, it looks like the deserializer is working correctly, but something is going wrong.

### Expected behavior

The list should be converted to a tensor without error.

### Environment

- jsonargparse 4.15.2
- Python 3.10.6

**Comment by mauvilsa** [comment]
Thank you for reporting! Will fix it as soon as possible.

**Comment by mauvilsa** [comment]
I created a pull request with the fix #197. It was a small bug but I also added a unit test to avoid regressions. @samvanstroud can you please test it out with lightning and maybe review the pull request? You can install directly from the branch as:

```
pip install "jsonargparse @ https://github.com/omni-us/jsonargparse/zipball/issue-196-non-bool-cast-type"
```

**Comment by samvanstroud** [comment]
Thanks a lot @mauvilsa! That is working perfectly for me, I don't have any comments on the implementation - looks clean.
